File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let set = new Set();
    for (let num of nums) {
        if (set.has(num)) {
            return true; // Duplicate found
        } else {
            set.add(num);
        }
    }
    return false; // No duplicates found
};

File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        // if(nums.size()<=1){
        //     return false;
        // }
        // for(int i=0;i<nums.size();i++){
        //     if(nums[i]==nums[nums.size()-i-1]){
        //         return true;
        //     }
        // }
        // return false;

        // TLE 

        // for(int i=0;i<nums.size();i++){
        //     for(int j=i+1;j<nums.size();j++){
        //         if(nums[i]==nums[j]){
        //             return true;
        //         }
        //     }
        // }
        // return false;

        std::set<int> s;
        for(int i=0;i<nums.size();i++){
            s.insert(nums[i]);
        }
        if(s.size()==nums.size()){
            return false;
        }
        return true;
    }
};